bash_completion

```diff
@@ -995,28 +1057,27 @@ _comp_initialize()
 # @return True (0) if an option was found, False (> 0) otherwise
 __parse_options()
 {
-    local option option2 i IFS=$' \t\n,/|'
+    local option option2 i
 
     # Take first found long option, or first one (short) if not found.
     option=
-    local reset=$(shopt -po noglob)
-    set -o noglob
-    local -a array=($1)
-    $reset
-    for i in "${array[@]}"; do
-        case "$i" in
-            ---*) break ;;
-            --?*)
-                option=$i
-                break
-                ;;
-            -?*) [[ $option ]] || option=$i ;;
-            *) break ;;
-        esac
-    done
+    local -a array
+    if _comp_split -F $' \t\n,/|' array "$1"; then
+        for i in "${array[@]}"; do
+            case "$i" in
+                ---*) break ;;
+                --?*)
+                    option=$i
+                    break
+                    ;;
+                -?*) [[ $option ]] || option=$i ;;
+                *) break ;;
+            esac
+        done
+    fi
     [[ $option ]] || return 1
 
-    IFS=$' \t\n' # affects parsing of the regexps below...
+    local IFS=$' \t\n' # affects parsing of the regexps below...
```

Comment:
I think we could just remove this -- the comment is still true, but we are no longer modifying `$IFS` _ourselves_  in a way that it'd break those, and we're not doing this for other similar cases across the codebase either.

Reply:
Thank you. I have applied it eaeb53e8.